SwiftCompilerSources/Sources/Optimizer/FunctionPasses/AllocBoxToStack.swift

```diff
@@ -359,12 +359,25 @@ private func createAllocStack(for allocBox: AllocBoxInst, flags: Flags, _ contex
     stackLocation = asi
   }
 
+  let deadEndBlocks = context.deadEndBlocks
   for destroy in getFinalDestroys(of: allocBox, context) {
     let loc = allocBox.location.asCleanup.withScope(of: destroy.location)
     Builder.insert(after: destroy, location: loc, context) { builder in
-      if  !unboxedType.isTrivial(in: allocBox.parentFunction), !(destroy is DeallocBoxInst) {
+      if deadEndBlocks.isDeadEnd(destroy.parentBlock) {
```

Comment:
This is not correct for the corner case the alloc_box is within an infinite loop, e.g.
```
bb1:
  %1 = alloc_box
  destroy_value %1
  br bb1
```
If we are not inserting the destroy_addr we will leak the value. 

I don't have a good idea how to handle this without replicating half of the ossa-lifetime-completion code. Maybe a simple solution is to just not promote alloc_box in dead-end infinite loops.

Anyway, this problem should be solved once we have complete lifetimes and OSSA throughout the pipeline.
Maybe also worth adding a TODO to remove this check once we have complete lifetimes